caml_z.c

```diff
@@ -398,7 +398,7 @@ static value ml_z_reduce(value r, mp_size_t sz, intnat sign)
 #if Z_USE_NATINT
   if (!sz) return Val_long(0);
   if (sz <= 1 && Z_LIMB(r)[0] <= Z_MAX_INT) {
-    if (sign) return Val_long(-Z_LIMB(r)[0]); 
+    if (sign) return Val_long(0-Z_LIMB(r)[0]);
```

Comment:
The actual reason for this warning is explained here [msdn](https://msdn.microsoft.com/en-us/library/4kh09110.aspx).